tesseract_command_language/include/tesseract_command_language/core/waypoint.h

```diff
@@ -191,15 +208,21 @@ class Waypoint
   }
 
   Waypoint()  // NOLINT
-    : waypoint_(std::make_unique<detail_waypoint::WaypointInner<uncvref_t<NullWaypoint>>>())
+    : waypoint_(nullptr)
   {
   }
```

Comment:
Same comment about moving this constructor to `private`

Reply:
I was not able to move these to private because the utility function toArchiveStringXML, fromArchiveStringXML, etc have to create a default constructable object.

Reply:
I see. My work-around was to serialize/de-serialize the type into a `std::vector`. Declaring a `std::vector` of a specific type does not require the use the default constructor and `boost` has built-in serialization for most STL containers. Depending on what you think of the value of privatizing the default constructor, a work-around like this might be worth the little bit of indirection